### PR TITLE
extend devMiddlewareOptions with config.devServer

### LINKED ISSUE #650: devServer options not used by server
I'm specifying some devServer options in Storybook's webpack config:

```
 Object.assign(storybookBaseConfig, {
    devServer: {
      headers: { 'Access-Control-Allow-Origin': '*' }
    }
  });
```

When running the storybook server, the headers are not included:

```
➜  component-library git:(master) ✗ curl -v http://localhost:6006/                                                                     09:57:18
*   Trying ::1...
* Connected to localhost (::1) port 6006 (#0)
> GET / HTTP/1.1
> Host: localhost:6006
> User-Agent: curl/7.49.1
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Powered-By: Express
< Content-Type: text/html; charset=utf-8
< Content-Length: 1228
< ETag: W/"4cc-bRBT7YfeEe5YUtIAIwEt+w"
< Date: Wed, 14 Dec 2016 08:57:29 GMT
< Connection: keep-alive
<

    <!DOCTYPE html>
```

### PR DESCRIPTION
**Pull request by moimikey**
# notes

since `devMiddlewareOptions` is pretty much the same as `devServer`, this simply allows a webpack config to naturally use `devServer` to override the default `devMiddlewareOptions`

fixes #650

### CODE REVIEW

**Comment by moimikey** [comment]
@arunoda @mnmtanish

**Comment by ndelangen** [comment]
I need to take a look at this, I wondering if there's a way to add tests for this?

**Comment by moimikey** [comment]
i'll give it a shot

**Comment by ndelangen** [comment]
Awesome! I'll try to review it regardless.

**Comment by moimikey** [comment]
^_~

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Thank you @moimikey !

**Comment by moimikey** [comment]
👍 🌈

File: src/server/middleware.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ export default function (configDir) {
     noInfo: true,
     publicPath: config.output.publicPath,
     watchOptions: config.watchOptions || {},
+    ...config.devServer,
```
Comment (wildeyes) [inline src/server/middleware.js]: Where can I pass the `config.devServer` to, to manipulate these configurations?

Comment (ndelangen) [reply src/server/middleware.js]: Should be in the webpack.config.js export I'm pretty sure.

**Pull request merged**